src/server/hook_func.c

```diff
@@ -5321,6 +5328,38 @@ mk_deferred_hook_info(int index, int event, long long int tid)
 	return info;
 }
 
+/**
+ * @brief
+ *		check if there is any new pending action
+ *		if yes remove the current one
```

Comment:
does not look like this function itself does any removal

Reply:
That was an initial design, Thanks for noticing. Now removed this line in https://github.com/openpbs/openpbs/pull/1992/commits/d54ce17afb4e9d6cd1a3da070504726722e60e5c